### PR TITLE
Merge master HEAD into openj9-staging

### PR DESCRIPTION
**Pull request by JasonFengJ9**
Conflicts:
```
	src/java.base/share/classes/com/sun/crypto/provider/CipherCore.java
```

Note: `8255557: Decouple GCM from CipherCore` https://github.com/ibmruntimes/openj9-openjdk-jdk/commit/55387a2d979e3fe83e8fbb3f900f013bf4775cbc moved GCM, this PR is to resolve OpenJDK merging conflict. Enabling native GCM will be a separated JCL patch PR.
In addition, `com/sun/crypto/provider/NativeGaloisCounterMode.java` need to be fixed to adopt new `FeedbackCipher.decryptFinal`:
```
int decryptFinal(byte[] cipher, int cipherOffset, int cipherLen,
                      byte[] plain, int plainOffset)
         throws IllegalBlockSizeException, ShortBufferException {
```

Signed-off-by: Jason Feng <fengj@ca.ibm.com>

### CODE REVIEW

**Comment by keithc-ca** [comment]
> Enabling native GCM will be a separate JCL patch PR.

I'd prefer to see that included in this PR.

File: closed/src/java.base/share/classes/com/sun/crypto/provider/NativeGaloisCounterMode.java [hunk 1]
```diff
@@ -467,13 +467,12 @@ int decrypt(byte[] in, int inOfs, int len, byte[] out, int outOfs) {
...
                      byte[] out, int outOfs)
-        throws IllegalBlockSizeException, AEADBadTagException,
-        ShortBufferException {
+        throws IllegalBlockSizeException, ShortBufferException {
         if (len < 0) {
             throw new ProviderException("Input length is negative");
         }
         if (len < tagLenBytes - ibuffer.size()) {
-            throw new AEADBadTagException("Input too short - need tag");
+           throw new IllegalBlockSizeException("Input too short - need tag");
```
Comment (keithc-ca) [inline closed/src/java.base/share/classes/com/sun/crypto/provider/NativeGaloisCounterMode.java:475]: Please don't change the indentation (here or line 508); it was correct before.

Comment (JasonFengJ9) [reply closed/src/java.base/share/classes/com/sun/crypto/provider/NativeGaloisCounterMode.java]: Added back a space.

**Comment by JasonFengJ9** [comment]
> Enabling native GCM will be a separate JCL patch PR.
I'd prefer to see that included in this PR.

That requires revisiting native GCM implementation, It doesn't seem a quick change. 
I think it can be deferred.

Added a commit `Replace AEADBadTagException with IllegalBlockSizeException` https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/322/commits/f70b272fea1883edae09ba7da6e04722f1429838 to allow JDK17 compilation.

**Comment by keithc-ca** [comment]
Please create an issue to track enabling native GCM again and add a link to it here.

**Comment by JasonFengJ9** [comment]
> Please create an issue to track enabling native GCM again and add a link to it here.

Sure, created https://github.com/ibmruntimes/openj9-openjdk-jdk/issues/323

**Comment by keithc-ca** [comment]
Jenkins compile zlinux jdknext

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**